src/FlywheelCore.sol

```diff
@@ -145,10 +145,14 @@ contract FlywheelCore is Auth {
 
     function _addStrategyForRewards(ERC20 strategy) internal {
         require(strategyState[strategy].index == 0, "strategy");
-        strategyState[strategy] = RewardsState({index: ONE, lastUpdatedTimestamp: block.timestamp.safeCastTo32()});
+        uint32 lastUpdatedTimestamp = block.timestamp.safeCastTo32();
+        strategyState[strategy] = RewardsState({index: ONE, lastUpdatedTimestamp: lastUpdatedTimestamp});
 
         allStrategies.push(strategy);
         emit AddStrategy(address(strategy));
+
+        // initialize the rewards accrual
+        flywheelRewards.getAccruedRewards(strategy, lastUpdatedTimestamp);
```

Comment:
this does not necessarily initialize the reward accrual, and can be triggered via an `accrue` call

Reply:
Maybe bad choice of words to describe it. It needs to initialize the internal state of `flywheelRewards` so the first call to `accrue` and `claimRewards` can be successful. 

I am preparing an update that adds an abstract function `initializeStrategy()`  the `IFlywheelRewards` / `BaseFlywheelRewards` contracts that should be called after or inside `addStrategyForRewards()`